tests/test_dagger.py

```diff
@@ -25,62 +29,146 @@ def test_beta_schedule():
 
 
 def test_traj_collector(tmpdir):
-    env = gym.make(ENV_NAME)
+    venv = vec_env.DummyVecEnv([lambda: gym.make(ENV_NAME)])
     robot_calls = 0
 
-    def get_random_act(obs):
+    def get_random_acts(obs):
         nonlocal robot_calls
         robot_calls += 1
-        return env.action_space.sample()
+        return [venv.action_space.sample() for _ in range(venv.num_envs)]
 
     collector = dagger.InteractiveTrajectoryCollector(
-        env=env, get_robot_act=get_random_act, beta=0.5, save_dir=tmpdir
+        venv=venv, get_robot_acts=get_random_acts, beta=0.5, save_dir=tmpdir
     )
     collector.reset()
-    zero_action = np.zeros((), dtype="int")
-    obs, rew, done, info = collector.step(zero_action)
-    assert rew != 0
-    assert not done
-    assert isinstance(info, dict)
+    zero_acts = np.zeros((venv.num_envs,), dtype="int")
+    obs, rews, dones, infos = collector.step(zero_acts)
+    assert rews != 0
+    assert not np.any(dones)
+    for info in infos:
+        assert isinstance(info, dict)
     # roll out ~5 episodes
     for i in range(999):
-        _, _, done, _ = collector.step(zero_action)
+        _, _, done, _ = collector.step(zero_acts)
         if done:
             collector.reset()
 
     # there is a <10^(-12) probability this fails by chance; we should be calling
     # robot with 50% prob each time
     assert 388 <= robot_calls <= 612
 
+    # All user/expert actions are zero. Therefore, all collected actions should be
+    # zero.
     file_paths = glob.glob(os.path.join(tmpdir, "dagger-demo-*.npz"))
     assert len(file_paths) >= 5
     trajs = map(dagger._load_trajectory, file_paths)
     nonzero_acts = sum(np.sum(traj.acts != 0) for traj in trajs)
     assert nonzero_acts == 0
 
 
-def make_trainer(tmpdir, beta_schedule=dagger.LinearBetaSchedule(1)):
-    env = gym.make(ENV_NAME)
-    env.seed(42)
+@pytest.fixture
+def venv():
+    return util.make_vec_env(ENV_NAME, 2)
+
+
+@pytest.fixture
+def expert_policy(venv):
+    return serialize.load_policy("ppo", EXPERT_POLICY_PATH, venv)
+
+
+@pytest.fixture(params=[True, False])
+def expert_trajs(request):
+    keep_trajs = request.param
+    if keep_trajs:
+        with open(EXPERT_ROLLOUTS_PATH, "rb") as f:
+            return pickle.load(f)
+    else:
+        return None
+
+
+def _build_dagger_trainer(tmpdir, venv, beta_schedule, expert_policy, expert_trajs):
+    del expert_policy
+    if expert_trajs is not None:
+        pytest.skip(
+            "DAggerTrainer does not use trajectories. "
+            "Skipping to avoid duplicate test."
+        )
     return dagger.DAggerTrainer(
-        env,
-        tmpdir,
-        beta_schedule,
-        optimizer_kwargs=dict(lr=1e-3),
+        venv,
+        scratch_dir=tmpdir,
+        beta_schedule=beta_schedule,
+        bc_kwargs={"optimizer_kwargs": dict(lr=1e-3)},
```

Comment:
Nit: I find it odd that there's two different ways of defining a dictionary used here.